src/SFML/Network/IpAddress.cpp

```diff
@@ -210,7 +225,16 @@ bool operator !=(const IpAddress& left, const IpAddress& right)
 ////////////////////////////////////////////////////////////
 bool operator <(const IpAddress& left, const IpAddress& right)
 {
-    return left.toInteger() < right.toInteger();
+    if (left.m_valid < right.m_valid)
+        return true;
+
+    if (right.m_valid < left.m_valid)
+        return false;
+
+    if (left.m_address < right.m_address)
+        return true;
+
+    return false;
 }
```

Comment:
So... roll our own or `std::make_pair`? There isn't that big of a difference unless you are using a really crappy compiler that doesn't know how to optimize out the pair constructions. I don't mind either of them and just chose the one with the guaranteed performance advantage.

@LaurentGomila, @Bromeon